includes/class-amp-comment-walker.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Class AMP_Comment_Walker
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_Comment_Walker
+ *
+ * Walker to wrap comments in mustache tags for amp-template.
+ */
+class AMP_Comment_Walker extends Walker_Comment {
+
+	/**
+	 * The original comments arguments.
+	 *
+	 * @since 0.7
+	 * @var array
+	 */
+	public $args;
+
+	/**
+	 * Ends the element output, if needed.
+	 *
+	 * @since 0.7
+	 *
+	 * @see Walker::end_el()
+	 * @see wp_list_comments()
+	 *
+	 * @param string     $output  Used to append additional content. Passed by reference.
+	 * @param WP_Comment $comment The current comment object. Default current comment.
+	 * @param int        $depth   Optional. Depth of the current comment. Default 0.
+	 * @param array      $args    Optional. An array of arguments. Default empty array.
+	 */
+	public function end_el( &$output, $comment, $depth = 0, $args = array() ) {
+		$output .= '<amp-comment data-sort-time="' . strtotime( $comment->comment_date ) . '"  />';
```

Comment:
I don't think an `amp-comment` element should be introduced here. Rather, the `data-sort-time` attribute should just be added to the existing element. So instead of overriding the `end_el` method, you should instead extend the protected `comment` method and do something like:

```php
protected function comment( $comment, $depth, $args ) {
    ob_start();
    parent::comment( $comment, $depth, $args );
    $html = ltrim( ob_get_clean() );
    $html = preg_replace( '/^(<\w+)(?=\s)/', '$1 data-sort-time="' . strtotime( $comment->comment_date ) . '" );
    echo $html;
}
```

By doing this, there may not be a need for `AMP_Comments_Sanitizer`, except for adding the `on` attribute to the `form#commentform` element.

Reply:
In that instance, we could do the same string replace for the comment form attribute.
I was hesitant to extend the comment function since there was a lot of code in it and it worries me overriding such large methods, but your solution there is elegant, so will do that. Thanks!